src/mon/ConfigMonitor.cc

```diff
@@ -542,6 +543,16 @@ bool ConfigMonitor::prepare_command(MonOpRequestRef op)
     name = ConfFile::normalize_key_name(name);
     
     if (prefix == "config set" && !force) {
+
+      std::regex pattern("[!@#$%^&*]");
```

Comment:
If all you want to do is search for one char of a given list - regexp search is the wrong choice.
Use something like
if (who.find_first_of("!@#...") != string::npos) ...